client/src/components/VendorDashboard/VendorDashboard.js

```diff
@@ -0,0 +1,33 @@
+import React, { Component, Fragment } from "react";
+import "antd/dist/antd.css";
+import { connect } from "react-redux";
+
+class VendorDashboard extends Component {
+  render() {
+    const { isAuthenticated, user } = this.props;
+    if (isAuthenticated && user.role === "vendor") {
+      return (
+        <Fragment>
+          <p>Vendor Logged in</p>
+        </Fragment>
+      )
+    }
+    else {
+      return (
+        <Fragment>
+          <p>Please login to view vendor Profile.</p>
+        </Fragment>
+      )
+    }
+  }
+}
+
+const mapStateToProps = state => ({
+  isAuthenticated: state.auth.isAuthenticated,
+  user: state.auth.vendor
```

Comment:
In redux vendor state is not there, there is user which has a property of isVendor.
I'll merge and fix.